LWA/LWA_bifrost.py

```diff
@@ -844,109 +942,149 @@ def main(self):
-                                    autocorr_lo = bifrost.ndarray(numpy.ones(shape=(3,1,nchan,nstand,npol**2),dtype=numpy.int32)*self.grid_size/2,space='cuda')
-                                    autocorr_il = bifrost.ndarray(numpy.ones(shape=(1,nchan,nstand,npol**2,self.ant_extent,self.ant_extent),dtype=numpy.complex64),space='cuda')
-
+                                    autocorrs = bifrost.ndarray(
+                                        shape=(self.ntime_gulp, nchan, nstand, npol ** 2),
+                                        dtype=numpy.complex64,
+                                        space="cuda",
+                                    )
+                                    autocorrs_av = bifrost.zeros(
+                                        shape=(1, nchan, nstand, npol ** 2),
+                                        dtype=numpy.complex64,
+                                        space="cuda",
+                                    )
+                                    autocorr_g = bifrost.zeros(
+                                        shape=(1, nchan, npol ** 2, self.grid_size, self.grid_size),
+                                        dtype=numpy.complex64,
+                                        space="cuda",
+                                    )
+                                    autocorr_lo = bifrost.ndarray(
+                                        numpy.ones(
+                                            (3, 1, nchan, nstand, npol ** 2),
+                                            dtype=numpy.int32
+                                        ) * self.grid_size / 2,
+                                        space="cuda",
+                                    )
+                                    autocorr_il = bifrost.ndarray(
+                                        numpy.ones(
+                                            (1, nchan, nstand, npol ** 2, self.ant_extent, self.ant_extent),
+                                            dtype=numpy.complex64
+                                        ),
+                                        space="cuda",
+                                    )
 
                                 # Cross multiply to calculate autocorrs
-                                bifrost.map('a(i,j,k,l) += (b(i,j,k,l/2) * b(i,j,k,l%2).conj())',
-                                            {'a':autocorrs, 'b':udata,'t':self.ntime_gulp},
-                                            axis_names=('i','j','k','l'),
-                                            shape=(self.ntime_gulp,nchan,nstand,npol**2))
-
-                            bifrost.map('a(i,j,p,k,l) += b(0,i,j,p/2,k,l)*b(0,i,j,p%2,k,l).conj()',
-                                        {'a':crosspol, 'b':gdata},
-                                        axis_names=('i','j', 'p', 'k', 'l'),
-                                        shape=(self.ntime_gulp, nchan, npol**2, self.grid_size, self.grid_size))
-
+                                bifrost.map(
```

Comment:
Like-wise you can replace this function call with following : 

                               ## Auto-Correlation of Antennas
                                if self.benchmark == True:
                                  timeacorr1 = time.time()

                                try:
                                  bf_acorr.execute(udata, autocorrs)
                                except NameError:
                                  bf_acorr = aCorr()
                                  bf_acorr.init(self.locs, polmajor=False)
                                  bf_acorr.execute(udata, autocorrs)

                                if self.benchmark == True:
                                  timeacorr2 = time.time()
                                  print(" Auto-Corr time : %f" % (timeacorr2 - timeacorr1))